File: YEAR2/SEM2/MPP/TemaUI/repository0/IParticipant.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TemaUI.domain;
using TemaUI.repository0;

namespace TemaUI.repository0
{
    public interface IParticipant: IRepository<long,Participant>
    {
        public int participantiprobe(String proba1);
        public int participantiCategorie(int m, int M, String probaq);

        public IEnumerable<Participant> participantlaProba(String proba);

        public IEnumerable<Participant> participantCategorie(int varstam, int varstaM, String proba);

        public Participant searchwithCNP(String CNP);
    }
}

File: YEAR2/SEM2/Medii de proiectare si programare/Tema1/src/main/java/org/example/repository/database/IParticipant.java
package org.example.repository.database;

import org.example.domain.Participant;
import org.example.repository.Repository0;

import java.util.List;

public interface IParticipant extends Repository0<Long, Participant> {

    int participantiprobe();

    int participantiCategorie();

    List<Participant> participantlaProba(String proba);

    List<Participant> participantCategorie(int varsta);

}
